internal/blocktx/peer_handler.go

```diff
@@ -328,6 +356,58 @@ func (ph *PeerHandler) startProcessTxs() {
 	}()
 }
 
+func (ph *PeerHandler) startProcessRequestTxs() {
+	ph.quitListenRequestTxChannel = make(chan struct{})
+	ph.quitListenRequestTxChannelComplete = make(chan struct{})
+	updatesBatch := make([]*blocktx_api.TransactionBlock, ph.registerRequestTxsBatchSize)
+
+	ticker := time.NewTicker(ph.registerRequestTxsInterval)
+	go func() {
+		defer func() {
+			ph.quitListenRequestTxChannelComplete <- struct{}{}
+		}()
+
+		for {
+			select {
+			case <-ph.quitListenRequestTxChannel:
+				return
+			case txHash := <-ph.requestTxChannel:
+				blockHash, blockHeight, merklePath, err := ph.store.GetMinedTransaction(context.Background(), txHash)
+				if err != nil {
+					continue
+				}
+
+				updatesBatch = append(updatesBatch, &blocktx_api.TransactionBlock{
+					TransactionHash: txHash,
+					BlockHash:       blockHash,
+					BlockHeight:     blockHeight,
+					MerklePath:      merklePath,
+				})
+
+				if len(updatesBatch) < ph.registerRequestTxsBatchSize || len(updatesBatch) == 0 {
+					continue
+				}
+
+				if err = ph.mqClient.PublishMinedTxs(updatesBatch); err != nil {
+					ph.logger.Error("failed to publish mined txs for requested hashes")
```

Comment:
You may want to continue after this error, otherwise you delete all items in updatesBatch before the publishing was successful